### PR TITLE
[ALLUXIO-3356] Respect hostname/port set in hdfs URI

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-3356

### CODE REVIEW

**Comment by aaudiber** [comment]
Jenkins, test this please

**Comment by aaudiber** [comment]
1. `connectDetailsMatch` does not mutate state. It creates a copy of the configuration with
    ```java
    AlluxioConfiguration alluxioConf = new InstancedConfiguration(Configuration.global());
    ```
    and operates on that.
2. We aren't modifying the global configuration

It is difficult to add a unit test for this because the only public method which uses this also calls
```java
    // Try to connect to master, if it fails, the provided uri is invalid.
    FileSystemMasterClient client = FileSystemContext.get().acquireMasterClient();
    try {
      client.connect();
      // Connected, initialize.
    } finally {
      FileSystemContext.get().releaseMasterClient(client);
    }
```

Even an integration test would be difficult because we would need two different clusters to be running, or at least have two different hostnames for the same cluster.

File: core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java [hunk 1]
```diff
@@ -537,6 +539,14 @@ private boolean connectDetailsMatch(URI uri, org.apache.hadoop.conf.Configuratio
     // Alluxio Configuration.
     AlluxioConfiguration alluxioConf = new InstancedConfiguration(Configuration.global());
```
Comment (gpang) [inline core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: is this a temporary conf? Is the host and port not needed elsewhere/later?

Comment (aaudiber) [reply core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: This method just determines whether the connection details implied by the URI + HDFS configuration are different from the current connection details used by the FileSystemContext singleton. If they don't match, the global configuration will be properly updated elsewhere

Comment (gpang) [reply core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: I see, so somewhere else, we already use `uri.getHost()` if it is there, and this method is just for checking? If that is the case, then shouldn't that other place update the conf correctly, to use the `uri.getHost()`?

**Comment by calvinjia** [comment]
Ah okay, I didn't realize there is an existing piece of code that does the update.

File: core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java (hunk 1 above)
Comment (aaudiber) [reply core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: we check for changes in one place and then make the changes in another

**Pull request merged**